src/container.ts

```diff
@@ -25,10 +26,14 @@ interface Exec {
   inspect(): Promise<ExecInspectResult>;
 }
 
+interface StopOptions {
```

Comment:
1. Expose the StopOptions/RemoveOptions interfaces
I created these interfaces for better readability, I didn't plan for them to be used by clients.

2. Exporting StartedTestContainer interfaces.
Good idea, I'll do the same for the StoppedTestContainer as well. shortly.

3. Propagating StopOptions/RemoveOptions.
Good catch, will fix that soon.

Reply:
https://github.com/testcontainers/testcontainers-node/pull/34

Reply:
Just to make sure I understand point 1, I'm trying to do the following:
```ts
import { GenericContainer, Wait, StartedTestContainer, StopOptions } from 'testcontainers';
....
let testContainer: StartedTestContainer;
// turn off durability to make test DB run faster
const testPostgresConfig = _.flatMap(['fsync', 'synchronous_commit', 'full_page_writes'], s => ['-c', `${s}=off`]);
const stopOptions: StopOptions = {
  // don't wait before killing the process, since we don't need to shutdown cleanly
  timeout: 0
};
....
testContainer = await new GenericContainer('postgres', '11.5-alpine')
  .withCmd(['postgres', ...testPostgresConfig])
  .withTmpFs({ [pgDataDir]: 'rw,noexec,nosuid,size=65536k' })
  .withEnv('POSTGRES_DB', databaseName)
  .withEnv('POSTGRES_USER', adminUser)
  .withEnv('POSTGRES_PASSWORD', adminPassword)
  .withEnv('PGDATA', pgDataDir)
  .withExposedPorts(internalDbPort)
  .withWaitStrategy(Wait.forLogMessage('listening on IPv4 address'))
   .withStartupTimeout(new Duration(120, TemporalUnit.SECONDS))
   .start();
....
// stop & remove the container, and delete the associated volume
await testContainer.stop(stopOptions);
```
Does this match how you expected it to be used?

Reply:
My thinking (which may be wrong here) is that the type `StopOptions` shouldn't be a concern of a client. So in your example you can define the object `stopOptions` of any shape, and the compiler will tell you if it conforms to what testcontainers expects.

Example:
```typescript
const stopOptions = { timeout: new Duration(10, TemporalUnit.SECONDS) }
container.stop(stopOptions) # compiles

const stopOptions = { timeout: "hi" }
container.stop(stopOptions) # does not compile
```

Note that the client doesn't need to worry about the type `StopOptions`.

I can't really say whether exposing it is right or wrong, but the compiler should be all the client needs here

Reply:
Right, I understand that Typescript can figure it out if I just pass a value, but I think it's normally good practice to export anything that's part of your public interface.

Reply:
I understand your point, but just because it has the word `interface` in it doesn't mean it's *public*. This is what I had:

```typescript
export interface StartedTestContainer {
  stop(options?: { 
    timeout?: Duration, 
    removeVolumes?: boolean 
  }): Promise<StoppedTestContainer>;
  ...
}
```

Which lends itself to my first example, which is how I thought it would be used.

I merely extracted the object to a type for readability, I didn't intend for this type to be exported as part of the public API

Reply:
Fair enough.